src/main/java/org/example/Book.java

```diff
@@ -0,0 +1,35 @@
+package org.example;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.time.LocalDate;
+
+@Getter
+@Setter
+public class Book {
+
+    private String title;
+    private Author author;
+    private LocalDate publication;
+    private int pages;
+    private String genre;
+
+    private Book(String title, Author author, LocalDate publication, Integer pages, String genre) {
+        this.title = title;
+        this.author = author;
+        this.publication = publication;
+        this.pages = pages;
+        this.genre = genre;
+    }
+
+    public static Book createBook(String title, Author author, LocalDate publication, Integer pages, String genre) {
+
+        if (title == null || author == null || publication == null || pages < 0 || genre == null) {
```

Comment:
kann ein publicationdate neuer sein als das aktuelle datum? (also in der zukunft liegen?

kann ein buch 0 seiten haben?

Reply:
Es kann doch auch in der Zukunft liegen oder nicht? Oft kann man Bücher preordern, da liegt das publicationdate auch in der Zukunft.

Reply:
einverstanden 👍